deploy/azure.tf

```diff
@@ -0,0 +1,192 @@
+#
+#   export ARM_SUBSCRIPTION_ID=<YOUR SUBSCRIPTION ID>
+#   export ARM_CLIENT_ID=<SERVICE PRINCIPAL ID>
+#   export ARM_CLIENT_SECRET=<SERVICE PRINCIPAL SECRET>
+#   export ARM_TENANT_ID=<ACTIVE DIRECTORY DOMAIN ID>
+#
+# or set up them here in this configuration file:
+#
+#   provider "azurerm" {
+#     subscription_id = "xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+#     client_id       = "xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+#     client_secret   = "xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+#     tenant_id       = "xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+#   }
+#
+# Then run this script with
+# ./terraform.exe plan -var-file=tfvars.json
+
+variable location {
+    type = "string"
+}
+
+variable "cosmosdb_failover_location" {
+    type = "string"
+}
+
+variable "azurerm_resource_group_00" {
+    type = "string"
+}
+
+variable "azurerm_storage_account_00" {
+    type = "string"
+} 
+
+variable "azurerm_storage_container_00" {
+    type = "string"
+}
+
+variable "azurerm_storage_queues" {
+    type    = "list"
+}
+
+variable "azurerm_cosmosdb_00" {
+    type = "string"
+}
+
+variable "azurerm_app_service_plan_00" {
+    type = "string"
+}
+
+variable "azurerm_application_insights_00" {
+    type = "string"
+}
+
+variable "azurerm_log_analytics_00" {
+    type = "string"    
+}
+
+# module "variables" {
+#     source = "./modules/variables"
+# }
+
+## RESOURCE GROUP
+
+# Create a resource group if it doesn’t exist
+resource "azurerm_resource_group" "azurerm_resource_group_00" {
+    name     = "${var.azurerm_resource_group_00}"
+    location = "${var.location}"
+    tags {
+        environment = "production"
+    }
+}
+
+## STORAGE
+
+resource "azurerm_storage_account" "azurerm_storage_account_00" {
+    name                = "${var.azurerm_storage_account_00}"
+    resource_group_name = "${azurerm_resource_group.azurerm_resource_group_00.name}"
+    location            = "${azurerm_resource_group.azurerm_resource_group_00.location}"
+
+    # can be one between Premium_LRS, Standard_GRS, Standard_LRS, Standard_RAGRS, Standard_ZRS
+    # see https://docs.microsoft.com/en-us/azure/storage/common/storage-redundancy
+    account_type = "Standard_GRS"
+
+    # see https://docs.microsoft.com/en-us/azure/storage/common/storage-service-encryption
+    enable_blob_encryption = true
+
+    tags {
+        environment = "production"
+    }
+}
+
+resource "azurerm_storage_container" "azurerm_storage_container_00" {
+    name                    = "${var.azurerm_storage_container_00}"
+    resource_group_name     = "${azurerm_resource_group.azurerm_resource_group_00.name}"
+    storage_account_name    = "${azurerm_storage_account.azurerm_storage_account_00.name}"
+
+    # Can be either blob (to publish blob on internet),container (to publish everything) or private
+    container_access_type   = "private"
+}
+
+## QUEUES
+
+resource "azurerm_storage_queue" "azurerm_storage_queue_00" {
+    name                 = "${var.azurerm_storage_queues[0]}"
```

Comment:
if you want you could [interpolate over the list of queues](https://www.terraform.io/docs/configuration/interpolation.html)

however, I prefer to be explicit and avoid creating "critical" resources from lists (i.e. `var.created_messages_queue.name`)